kubeflow-pipelines.yaml

```diff
@@ -0,0 +1,132 @@
+          # Mitigate GHSA-8c26-wmh5-6g9v
+          go get golang.org/x/crypto@1baeb1ce4c0b
+
+          # GHSA-p782-xgp4-8hr8
+          go get golang.org/x/sys@v0.8.0
+          go mod tidy
+
+          mkdir -p ${{targets.subpkgdir}}/usr/bin
+          mkdir -p ${{targets.subpkgdir}}/third_party
+
+          name=$(echo "${{range.value}}" | cut -d ';' -f 1)
+          path=$(echo "${{range.value}}" | cut -d ';' -f 2-)
+
+          make build/$name
+          install -Dm644 ./build/$name ${{targets.subpkgdir}}/usr/bin/${{range.key}}
+
+          go-licenses check ./src/$path
+          go-licenses csv ./src/$path > ${{targets.subpkgdir}}/third_party/$name.csv
+          go-licenses save ./src/$path --save_path ${{targets.subpkgdir}}/third_party/NOTICES
+
+          if [ "${name}" == "apiserver" ]; then
+            mkdir -p ${{targets.subpkgdir}}/samples
+            pip3 install --prefer-binary -r requirements.txt --prefix=/usr --root=${{targets.subpkgdir}}
+            cp -r ../samples/* ${{targets.subpkgdir}}/samples
+            cp ./src/$path/config/sample_config.json ${{targets.subpkgdir}}/samples/sample_config.json
+          fi
+
+  - name: "kubeflow-pipelines-cache-deployer"
+    description: "Kubeflow Pipelines cache-deployer"
+    pipeline:
+      - runs: |
+          mkdir -p ${{targets.subpkgdir}}/usr/bin
+          mkdir -p ${{targets.subpkgdir}}/third_party
+          install -m755 -D third_party/license.txt ${{targets.subpkgdir}}/third_party/license.txt
+          install -m755 -D backend/src/cache/deployer/deploy-cache-service.sh ${{targets.subpkgdir}}/usr/bin/deploy-cache-service.sh
+          install -m755 -D backend/src/cache/deployer/webhook-create-signed-cert.sh ${{targets.subpkgdir}}/usr/bin/webhook-create-signed-cert.sh
+          install -m755 -D backend/src/cache/deployer/webhook-patch-ca-bundle.sh ${{targets.subpkgdir}}/usr/bin/webhook-patch-ca-bundle.sh
+
+  - name: "kubeflow-pipelines-cache-deployer-compat"
+    dependencies:
+      runtime:
+        - kubeflow-pipelines-cache-deployer
+    pipeline:
+      - runs: |
+          mkdir -p ${{targets.subpkgdir}}/kfp/cache/deployer/
+          ln -s /usr/bin/deploy-cache-service.sh "${{targets.subpkgdir}}"/kfp/cache/deployer/deploy-cache-service.sh
+          ln -s /usr/bin/webhook-create-signed-cert.sh "${{targets.subpkgdir}}"/kfp/cache/deployer/webhook-create-signed-cert.sh
+          ln -s /usr/bin/webhook-patch-ca-bundle.sh "${{targets.subpkgdir}}"/kfp/cache/deployer/webhook-patch-ca-bundle.sh
+
+  - name: "kubeflow-pipelines-frontend-compat"
+    description: "Kubeflow Pipelines frontend"
+    pipeline:
+      - runs: |
+          mkdir -p ${{targets.subpkgdir}}/server
+          mkdir -p ${{targets.subpkgdir}}/client
+          cd frontend/
+          npm ci
+          npm run postinstall
+          npm run build
+          cp -r ./server/* ${{targets.subpkgdir}}/server
+          cp -r ./build/* ${{targets.subpkgdir}}/client
```

Comment:
should this be in the main pipeline and the compat subpackage just be symlinks?  I'm thinking that by default we'd want everything to work using our standard folder structure and use a compat subpackage for symlinks where there are non standard folder layouts.

Reply:
Definitely agree with you. But in this case, I couldn't find a way out since `npm` outputs multiple files under both of `/server/` and `/build/` folders. There is no single binary that we can put under `/usr/bin/`. On the other hand, creating a folder under `/usr/bin/` would look weird. How should we proceed here?

Reply:
could we have both a frontend and compat package?

i.e. something like

```yaml
  - name: "kubeflow-pipelines-frontend"
    description: "Kubeflow Pipelines frontend"
    pipeline:
      - runs: |
          mkdir -p ${{targets.subpkgdir}}/usr/bin
          cd frontend/
          npm ci
          npm run postinstall
          npm run build
          cp -r ./server/* ${{targets.subpkgdir}}/usr/bin/
          cp -r ./build/* ${{targets.subpkgdir}}/usr/bin/
```
and then the compat

```yaml
  - name: "kubeflow-pipelines-frontend-compat"
    description: "Kubeflow Pipelines frontend compatibility package"
    pipeline:
      - runs: |
          mkdir -p ${{targets.subpkgdir}}/server
          mkdir -p ${{targets.subpkgdir}}/client

          ln -sf /usr/bin ${{targets.subpkgdir}}/server
          ln -sf /usr/bin ${{targets.subpkgdir}}/client
```

I wonder if something like that would work?  This way we are installing the binaries into the standard wolfi location and using symlinks to support upstream non standard folders?